### ISSUE TITLE
allocation_tracer: no member named 'flags' in 'struct RBasic'

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1978]
```
truffleruby 20.1.0-dev-abe0c051, like ruby 2.6.5, GraalVM CE Native [x86_64-darwin]
```

To reproduce

```
gem install allocation_tracer
```

Error:

```
Building native extensions. This could take a while...
ERROR:  Error installing allocation_tracer:
	ERROR: Failed to build gem native extension.

    current directory: ~/.gem/truffleruby/2.6.5/gems/allocation_tracer-0.6.3/ext/allocation_tracer
~/.rubies/truffleruby-dev/bin/truffleruby -I ~/.rubies/truffleruby-dev/lib/mri -r ./siteconf20200326-72649-15wpr0x.rb extconf.rb
creating Makefile

current directory: ~/.gem/truffleruby/2.6.5/gems/allocation_tracer-0.6.3/ext/allocation_tracer
make "DESTDIR=" clean

current directory: ~/.gem/truffleruby/2.6.5/gems/allocation_tracer-0.6.3/ext/allocation_tracer
make "DESTDIR="
compiling allocation_tracer.c
allocation_tracer.c:248:32: error: no member named 'flags' in 'struct RBasic'
    info->flags = RBASIC(obj)->flags;
                  ~~~~~~~~~~~  ^
allocation_tracer.c:400:29: error: no member named 'flags' in 'struct RBasic'
        info->flags = RBASIC(obj)->flags;
                      ~~~~~~~~~~~  ^
allocation_tracer.c:625:29: error: no member named 'flags' in 'struct RBasic'
        info->flags = RBASIC(obj)->flags;
                      ~~~~~~~~~~~  ^
3 errors generated.
make: *** [allocation_tracer.o] Error 1

make failed, exit code 2

Gem files will remain installed in ~/.gem/truffleruby/2.6.5/gems/allocation_tracer-0.6.3 for inspection.
Results logged to ~/.gem/truffleruby/2.6.5/extensions/x86_64-darwin/20.1.0-dev-abe0c051/allocation_tracer-0.6.3/gem_make.out
```

**Comment by eregon** [comment]
Thank you for the report, this is the same error as in #1891.

**Comment by norswap** [comment]
I'll push a fix for this `RBasic->flags` soon, but now the gem will not install because we don't support `RUBY_INTERNAL_EVENT_NEWOBJ` and `RUBY_INTERNAL_EVENT_FREEOBJ` (yet).

**Comment by eregon** [comment]
I'll take care of the RUBY_INTERNAL_EVENT_NEWOBJ/RUBY_INTERNAL_EVENT_FREEOBJ issue.
That conflicts with our fix for https://github.com/oracle/truffleruby/issues/1983

So I think now we should define them, but adding a TracePoint for those would not do anything, just warn.

**Comment by norswap** [comment]
Fixed by b9a3868f1077e3f4f91f7fad15fc536fb138747f and  9ee88a5a34f9481b0ee2f63fb56eb150fab87766
